utoipa-swagger-ui/build.rs

```diff
@@ -22,34 +22,40 @@ use zip::{result::ZipError, ZipArchive};
 const SWAGGER_UI_DOWNLOAD_URL_DEFAULT: &str =
     "https://github.com/swagger-api/swagger-ui/archive/refs/tags/v5.17.3.zip";
 
+const SWAGGER_UI_DOWNLOAD_URL: &str = "SWAGGER_UI_DOWNLOAD_URL";
+const SWAGGER_UI_OVERWRITE_FOLDER: &str = "SWAGGER_UI_OVERWRITE_FOLDER";
+
 fn main() {
     let target_dir = env::var("OUT_DIR").unwrap();
-    println!("OUT_DIR: {}", target_dir);
+    println!("OUT_DIR: {target_dir}");
 
     let url =
-        env::var("SWAGGER_UI_DOWNLOAD_URL").unwrap_or(SWAGGER_UI_DOWNLOAD_URL_DEFAULT.to_string());
+        env::var(SWAGGER_UI_DOWNLOAD_URL).unwrap_or(SWAGGER_UI_DOWNLOAD_URL_DEFAULT.to_string());
```

Comment:
`url` is parsed into `url::Url` on every branch taken (explicitly for file://, explicitly for curl, and implicitly via `IntoUrl` for reqwest).

Doing it up here instead would also give access to `Url::schema` instead of approximating it with `str::starts_with`.

Reply:
Indeed.

Reply:
True that. Maybe something that can be considered in future if it starts to bother someone :laughing: Not going to edit it for now though